pkg/tracing/logger.go

```diff
@@ -195,6 +214,12 @@ func (n consoleLogger) StartSpan(ctx context.Context, op string, attrs ...attrib
 	return NewScopedLogger(childCtx, childSpan, n)
 }
 
+func (n consoleLogger) StartSpanNamedByCaller(ctx context.Context, attrs ...attribute.KeyValue) (context.Context, ScopedLogger) {
```

Comment:
This shouldn't re-implement StartSpan and instead we can reuse StartSpan and supply the op name from the caller

```
return n.StartSpan(ctx, callerFunc(0), attrs...)
```